portal-api/src/v1/gef/controllers/facilities.controller.js

```diff
@@ -170,18 +207,25 @@ exports.updatePUT = async (req, res) => {
     return res.status(422).send(enumValidationErr);
   }
 
-  let response;
-  const updatedFacility = await update(req.params.id, req.body, generatePortalAuditDetails(req.user._id));
+  try {
+    let response;
+    const updatedFacility = await update(req.params.id, req.body, generatePortalAuditDetails(req.user._id));
+
+    if (updatedFacility.value) {
+      response = {
+        status: facilitiesStatus(updatedFacility.value),
+        details: updatedFacility.value,
+        validation: facilitiesValidation(updatedFacility.value),
+      };
+    }
 
-  if (updatedFacility.value) {
-    response = {
-      status: facilitiesStatus(updatedFacility.value),
-      details: updatedFacility.value,
-      validation: facilitiesValidation(updatedFacility.value),
-    };
+    return res.status(utils.mongoStatus(updatedFacility)).send(response);
+  } catch (error) {
+    if (error instanceof ApiError) {
+      return res.status(error.status).send({ status: error.status, message: error.message });
+    }
+    throw error;
   }
-
-  return res.status(utils.mongoStatus(updatedFacility)).send(response);
 };
```

Comment:
I'm thinking maybe wrapping the part that will error with an api error is probably better -- what do you think?
```suggestion
  let response;

  try {
    const updatedFacility = await update(req.params.id, req.body, generatePortalAuditDetails(req.user._id));
  } catch (error) {
    if (error instanceof ApiError) {
      return res.status(error.status).send({ status: error.status, message: error.message });
    }
    throw error;
  }

  if (updatedFacility.value) {
    response = {
    status: facilitiesStatus(updatedFacility.value),
    details: updatedFacility.value,
    validation: facilitiesValidation(updatedFacility.value),
    };
  }

  return res.status(utils.mongoStatus(updatedFacility)).send(response);
};
```

Reply:
`updatedFacility ` needs to be declared with a `let` outside the try block. A bit nyeh on that but happy to do if you prefer

Reply:
i think so :) `let` outside the try block is fine i think